src/rhash/hash.c

```diff
@@ -2388,23 +2380,202 @@ static int rc_hash_gamecube(char hash[33], const rc_hash_iterator_t* iterator)
   {
     if (dol_sizes[ix] == 0)
       continue;
-
+    rc_file_seek(iterator, file_handle, part_offset + dol_offsets[ix], SEEK_SET);
     if (ix < 7)
       rc_hash_iterator_verbose_formatted(iterator, "Hashing %u byte main.dol code segment %u", dol_sizes[ix], ix);
     else
       rc_hash_iterator_verbose_formatted(iterator, "Hashing %u byte main.dol data segment %u", dol_sizes[ix], ix - 7);
+    remaining_size = dol_sizes[ix];
+    while (remaining_size > READ_CHUNK_SIZE)
```

Comment:
Shouldn't this just be `remaining_size > dol_buf_size` as that's what was allocated?